app/controllers/posts_controller.rb

```diff
@@ -20,7 +20,10 @@ def create
   private
 
   def timeline_posts
-    @timeline_posts ||= Post.all.ordered_by_most_recent.includes(:user)
+    @requestee_friends = Friendship.where(requestor_id: current_user, status: 'accepted').pluck(:requestee_id)
+    @requester_friends = Friendship.where(requestee_id: current_user, status: 'accepted').pluck(:requestor_id)
+    @me_and_friends = @requestee_friends + @requester_friends + [current_user[:id]]
+    @timeline_posts ||= Post.where(user_id: @me_and_friends).ordered_by_most_recent.includes(:user)
   end
```

Comment:
Considering the “Fat Model, Skinny Controller” approach, this refers to how the M and C parts of MVC ideally work together. Namely, any non-response-related logic should go in the model, ideally in a nice, testable method. Meanwhile, the “skinny” controller is simply a nice interface between the view and model.
* Move the logic into your model and create a method, say `#me_and_friends_posts`, then simply call it from your controller like `Post.me_and_friends_posts`

Note this change is OPTIONAL but it will greatly improve your code